[RUN-3373] Move deployment settings in their own tab (for 10.3.0)

atennapel opened the pull request:
- The deployment settings have been moved from the "General" tab to a new tab "Deployment"
- Also fixes some incorrect numbering

atennapel-mx committed:
Specify custom repository settings

atennapel-mx committed:
Remove mention of custom repositories

Yiyun333 committed:
Merge branch 'development' into run/3373-custom-repos-for-gradle-10.3

Yiyun333 reviewed (commented):
Just merged development branch into this PR to solve some conflicts. As a result, the content of this PR is a bit messy now. I'll continue reviewing this PR after 10.3 is freezed.

Yiyun333 committed:
Align with the settings in the Preferences dialog box in 10.3

Yiyun333 committed:
Add anchor

Yiyun333 committed:
Update screenshot and small fix

Yiyun333 commented on content/en/docs/refguide/modeling/menus/edit-menu/preferences-dialog.md:
```diff
@@ -106,65 +112,87 @@ For example, you can see how to set up file comparison for Visual Studio Code be
 Select this option when you want to work on an app that is not stored in [Mendix Team Server](/developerportal/general/team-server/), but in a private Git server to which you have access. This will allow you to specify the location of the app on the Git server when opening, downloading, or uploading the app. In this section, you also need to specify name and email values that will be used to iden ...
 
-## 6 Work Environment Tab
+#### 6.2.6 Enable Automatic Fetching from a Remote Repository
+
+{{% todo %}}
+Add description
+{{% / todo %}}
+
+#### 6.2.7 Fetch Interval, Minutes
+
+{{% todo %}}
+Add description
+{{% / todo %}}
+
+### 6.3 Changes
+
+#### 6.3.1 Truncate Long Values
+
+{{% todo %}}
+Add description
+{{% / todo %}}
+
+### 6.4 Solution Warning
+
+#### 6.4.1 Show Warning on Updating Marketplace Modules
+
+{{% todo %}}
+Add description
+{{% / todo %}}
```
@atennapel I noticed that there are several settings missing for the **Git** section. I just added the structure here. Will you be able to add descriptions for these settings or do you know whom I can contact for these missing descriptions? Thank you!

atennapel replied:
I believe the MCR team manage that feature.

Yiyun333 replied:
@atennapel Thanks for the information! I will verify with the team.

Yiyun333 replied:
I am not going to update these sections in this PR. Hence, I've removed these sections from this PR in [this](https://github.com/mendix/docs/pull/6435/commits/2afc66e355075b82df4e082528341d879a5a90dd) commit. @MarkvanMents will take care of the update in the PR: https://github.com/mendix/docs/pull/6487/files.

Yiyun333 commented on content/en/docs/refguide/modeling/menus/edit-menu/preferences-dialog.md:
```diff
@@ -106,65 +112,87 @@ For example, you can see how to set up file comparison for Visual Studio Code be
 * `{0}` – this is replaced with the name of the original file before the arguments are passed to the file comparison program
 * `{1}` – this is replaced with the name of the changed file before the arguments are passed to the file comparison program
 
-### 5.2 Subversion 
-
-#### 5.2.1 Enable Private Version Control with Subversion {#enable}
-
-Select this option when you want to work on an app that is not stored in [Mendix Team Server](/developerportal/general/team-server/), but in another Subversion server to which you have access. This will allow you to specify the location of the app on the Subversion server when opening, downloading, or uploading the app.
```
I deleted this section because this setting is not in the **Preferences** dialog box anymore. Please let me know if you have different opinions.

atennapel replied:
This part I did not change from the UI...so I cannot say. Must have been the MCR team.

Yiyun333 replied:
@atennapel No problem. Will verify with them as well.

Yiyun333 replied:
Subversion was removed since 10 GA, so this deletion is fine. I am marking this as resolved.

Yiyun333 commented on content/en/docs/refguide/modeling/menus/edit-menu/preferences-dialog.md:
```diff
@@ -41,62 +41,68 @@ The **Auto-check delay** is the number of milliseconds that Studio Pro waits aft
+#### 3.2.2 Use Custom Repositories
 
-### 2.4 Disk Location
+Enabling this option allows you to specify which repositories should be used by Gradle.
 
-#### 2.4.1 Default App Directory {#default-directory}
+#### 3.2.3 Repositories
 
-This is the directory where new apps are stored. It is used in the [App Settings](/refguide/new-app/#app-settings) dialog box when you create a new app. If a different directory is selected in the **App Settings** dialog box, that directory is stored as the new default app directory.
+Here you can specify which repositories to use for Gradle. The content of this field should be specified using Groovy syntax and is what is inside the `repositories { }` section in a Gradle build file. By default, this field contains: `mavenCentral()`.
```
@atennapel I saw that you had added these two sections and then deleted them. I added them back again because they are in the **Preferences** dialog box in 10.3. Please let me know if you don't agree.

atennapel replied:
Indeed. At the time that I made this PR it was not certain if this setting would be expose in 10.3, but by now the decision has been made and indeed the Custom repos documentation should be added.

Yiyun333 committed:
Add and correct the "Enable private version control" anchor

Yiyun333 commented on content/en/docs/refguide/modeling/menus/version-control-menu/download-from-version-control-dialog.md:
```diff
@@ -23,7 +23,7 @@ You can also use the [Open App Dialog](/refguide/open-app-dialog/) to download a
 ## 2 Where Is Your App Stored?
 
-If **Enable private version control with Git** is set in the app [Preferences](/refguide/preferences-dialog/#enable), you can choose between the **Mendix Team Server** or a **Private server**. If it is not enabled, you will only be able to choose an app from the Mendix Team Server.
+If **Enable private version control with Git** is set in the app [Preferences](/refguide/preferences-dialog/#enable-with-Git), you can choose between the **Mendix Team Server** or a **Private server**. If it is not enabled, you will only be able to choose an app from the Mendix Team Server.
```
The old anchor for the **Enable Private Version Control with Subversion** setting doesn't seem to apply anymore. I think that the anchor should be placed for this **Enable Private Version Control with Git** section now. Hence, I added a new anchor for that section and also updated two version-control docs that used the old anchor. 

@MarkvanMents Can you please have a look at my update here? Thanks!

Yiyun333 commented on content/en/docs/refguide/modeling/menus/version-control-menu/upload-to-version-control-dialog.md:
```diff
@@ -41,7 +41,7 @@ You can only upload to an existing repository if the repository is currently emp
 ### 2.3 Private Server
 
-This option is only available when support for other servers is enabled in **Edit** > **Preferences** > **Advanced** > [Enable private version control](/refguide/preferences-dialog/#enable)).
+This option is only available when support for other servers is enabled in **Edit** > **Preferences** > **Advanced** > [Enable private version control](/refguide/preferences-dialog/#enable-with-Git)).
```
@MarkvanMents And here as well.

Yiyun333 committed:
Remove these to-dos in this PR

atennapel commented:
Note: we are working on a documentation page about the new Java dependency management feature (https://github.com/mendix/docs/pull/6561). It would be good to add some kind of link in the Custom repositories section to this new page (content/en/docs/refguide/java-programming/managed-dependencies.md), as we give more information about these repositories there.

Yiyun333 commented:
> Note: we are working on a documentation page about the new Java dependency management feature (#6561). It would be good to add some kind of link in the Custom repositories section to this new page (content/en/docs/refguide/java-programming/managed-dependencies.md), as we give more information about these repositories there.

@atennapel Can you please add references to this page with this link (/refguide/managed-dependencies/) to the Custom repositories section? I am not certain what exact information is wanted there. I'll review the link and the content after you've added something there. Thanks!

atennapel-mx committed:
Add link to the Managed Dependencies page

atennapel commented:
Yes, I just added a link in section 3.2.2. I was not sure if I should add it in 3.2.2 or 3.2.3 as it pertains to both sections, so I added it to the first one.

Yiyun333 commented:
> Yes, I just added a link in section 3.2.2. I was not sure if I should add it in 3.2.2 or 3.2.3 as it pertains to both sections, so I added it to the first one.

Thanks for adding the link! The location seems to be fine for now.
I just checked the other PR with the new page. I see that there is a section in that new page called **Custom Repositories**. Is that specific section that we want to point to instead of the whole page? If yes, I will add an anchor for that section and then point to that section. Currently, that PR needs a review first, so it's not in a big hurry for now. I can do this later after you've confirmed it.

atennapel commented:
Yes that is the correct section.

Yiyun333 commented:
Great! I will update the link and refer to that section when the review of the other PR is almost done! (Don't want to add an anchor there atm. A lot of commits there are still going on :)).

Yiyun333 committed:
Update preferences-dialog.md

Yiyun333 committed:
Update the cross-reference for custom repos
